### ISSUE TITLE
Subtitles

### ISSUE THREAD

**Issue by acoollevel** [acoollevel/uoa-mediaplayer-plus#39]
Auto-generated subtitles - unfortunately the only way to implement this would be to generate live in-browser subtitles as we can't store them on a server for copyright reasons.

**Comment by obwan02** [comment]
Would it still be legal if we used an instance of Mozilla's DeepSpeech https://github.com/mozilla/DeepSpeech on a server and _streamed_ the live audio to the server? 

If only parts of the audio of the lecture were being processed on the server at any time, and as soon as they were processed they were erased from the server then it wouldn't break the copyright?

**Comment by LRTNZ** [comment]
> 
> 
> Would it still be legal if we used an instance of Mozilla's DeepSpeech https://github.com/mozilla/DeepSpeech on a server and _streamed_ the live audio to the server?
> 
> If only parts of the audio of the lecture were being processed on the server at any time, and as soon as they were processed they were erased from the server then it wouldn't break the copyright?

Nope - we spoke to the people at UoA last year, and due to the fact the material is DRM'd, we cannot do that. It would be breaking the copyright.

**Comment by acoollevel** [comment]
The only way would be to run DeepSpeech locally via WebAssembly. However, automatically generated subtitles tend to be _not great_, without the ability for people to submit fixes. To my knowledge, the university has investigated officially supporting subtitles/transcriptions.

**Comment by LRTNZ** [comment]
Yeah, I think we talked about that a few months back @acoollevel. From memory, UoA was looking at getting a company to do the transcription work for them.
I think one of the big issues we found with doing it locally, apart from the lack of user corrections, would have been interpreting speech with a large number of potential accents. As UoA has such a broad range of accents among the teaching staff, we would not be able to use one "standard NZ accent" model, to do STT.

**Comment by obwan02** [comment]
It could be possible to train have an individual model for each lecturer, but yeah it kinda seems like a mish. And also there aren't any good in browser speech-to-text apis. Kinda frustrating but oh well.

**Comment by LRTNZ** [comment]
> 
> 
> It could be possible to train have an individual model for each lecturer, but yeah it kinda seems like a mish. And also there aren't any good in browser speech-to-text apis. Kinda frustrating but oh well.

Down the line, if Google made their live STT stuff on their pixel devices available, that would be amazing. 
But yeah, realistically, who has the time to train a model for each lecturer? Also, the lecturers could be different across recordings from a course.
So hopefully, UoA does get ontop of this issue asap.
